apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -313,6 +313,18 @@ type DatacenterOptions struct {
 	// a user-provided monitoring solution (at present, only support for Prometheus is available).
 	// +optional
 	Telemetry *telemetryapi.TelemetrySpec `json:"telemetry,omitempty"`
+
+	// AdditionalContainers defines additional containers to be deployed in each Cassandra pod.
+	// +optional
+	AdditionalContainers []corev1.Container `json:"additionalContainers,omitempty"`
+
+	// AdditionalInitContainers defines additional init containers to be deployed in each Cassandra pod.
+	// +optional
+	AdditionalInitContainers []corev1.Container `json:"additionalInitContainers,omitempty"`
+
+	// AdditionalVolumes defines additional volumes to be added to each Cassandra pod.
+	// +optional
+	AdditionalVolumes []cassdcapi.AdditionalVolumes `json:"additionalVolumes,omitempty"`
```

Comment:
I would prefer we handle volumes separately. I think it merits some more thought and discussion. AdditionalVolumes in cass-operator only supports PVs which are added to the PVC template of the StatefulSet. A volume mount is added in the cassandra container. Are we sure this is the behavior we want? Maybe it is, but I would rather take the time to discuss and be certain before we commit to the API change.

Reply:
sounds good, let's discuss this before moving forward with that specific part.